documentation/external_ssl_termination.markdown

```diff
@@ -5,21 +5,61 @@ canonical: "/puppetserver/latest/external_ssl_termination.html"
-
-> **Note**: Ruby Puppet master code uses the `ssl_client_header` and `ssl_client_verify_header` values to authorize the request per the rules in `auth.conf`. The `X-Client-Cert` value is only used for distilling trusted facts from extensions on the client certificate. To minimize the size of the request payload, you might want to omit the `X-Client-Cert` header from the request if trusted facts are not being used.
+## Disable HTTPS for Puppet Server
 
 You'll need to turn off SSL and have Puppet Server use the HTTP protocol instead: remove the `ssl-port` and `ssl-host` settings from the `conf.d/webserver.conf` file and replace them with `port` and `host` settings. See [Configuring the Webserver Service](https://github.com/puppetlabs/trapperkeeper-webserver-jetty9/blob/master/doc/jetty-config.md) for more information on configuring the web server service.
 
-To allow Puppet Server to recognize the HTTP headers described, open `config.d/master.conf` and add  `allow-header-cert-info: true` to the `webserver` config block. See [Puppet Server Configuration](./configuration.markdown) for more information on the `master.conf` file. Without `allow-header-cert-info` set to 'true', none of these HTTP headers are recognized by Puppet Server.
+## Allow Client Cert Data From HTTP Headers
+
+When using external SSL termination, Puppet Server expects to receive client certificate information via some HTTP headers.
 
-You'll need to reboot Puppet Server for the changes to take effect. The `ssl_client_header` and `ssl_client_verify_header` options in the `puppet.conf` file will then be enabled and work exactly as documented in Ruby Puppet. Please see the [Puppet Documentation](https://docs.puppetlabs.com/references/3.7.latest/configuration.html#sslclientheader) for more detail on these.
+By default, reading this data from headers is disabled. To allow Puppet Server to recognize it, open `config.d/master.conf` and add  `allow-header-cert-info: true` to the `webserver` config block. See [Puppet Server Configuration](./configuration.markdown) for more information on the `master.conf` file.
 
 > **WARNING**: Setting `allow-header-cert-info` to 'true' puts Puppet Server in an incredibly vulnerable state. Take extra caution to ensure it is **absolutely not reachable** by an untrusted network.
+>
+> With `allow-header-cert-info` set to 'true', core Ruby Puppet application code will use only the client HTTP header values---not an SSL-layer client certificate---to determine the client subject name, authentication status, and trusted facts. This is true even if the web server is hosting an HTTPS connection. This applies to validation of the client via rules in the [auth.conf](https://docs.puppetlabs.com/guides/rest_auth_conf.html) file and any [trusted facts](https://docs.puppetlabs.com/puppet/latest/reference/lang_facts_and_builtin_vars.html#trusted-facts) extracted from certificate extensions.
```

Comment:
You've got this trusted facts URL already defined as `[trusted]` below - might as well reuse it here